File: algos_assignment/3rd_grade/이서진_metamong/2nd_week1/10773.py
import sys
def push(x):
    stack.append(x)
def pop():
    if len(stack)==0:
        return -1
    else:
        return stack.pop()
n=int(sys.stdin.readline().rstrip())
sum=0
stack=[]
for i in range(n):
    num=int(sys.stdin.readline())
    if num==0:
        pop()
    else:
        push(num)
for i in stack :
    sum+=i
print(sum)

File: algos_assignment/team_pratice/ACM ICPC/코드알고싶조/10월 1주차/10773.cpp
#include <iostream>
#include <stack>
using namespace std;

int main()
{
	int k, n;
	cin >> k;
	stack <int> num;

	for (int i = 0; i < k; i++)
	{
		cin >> n;

		if (n == 0)
			num.pop();
		else
			num.push(n);
	}

	int sum = 0;
	while (!num.empty())
	{
		sum += num.top();
		num.pop();
	}

	cout << sum;
}
